site/content/entry/citra-progress-report-2018-Q1.md

```diff
@@ -0,0 +1,245 @@
+(for eg. Lighting LUTs).
+* 3) Copy the data to an internal staging buffer and ask the GPU to copy it again
+to the destination when it is ready. Twice the memory bandwidth, and switching
+between rendering and copy also takes a while for the GPU. This bottlenecks you
+hard if you do this per draw call.
+
+But in the end, all of them are bad in terms of performance. It is the simple
+upload call, but it either requires many copies and context switches, or it stalls
+the GPU.(_More on that in [here](https://de.slideshare.net/CassEveritt/approaching-zero-driver-overhead)_).
+So our buffer streaming way is a bit different. Now, we allocate a big
+staging buffer ourself, copy the data to the current position within the buffer,
+and tell the GPU to directly access the data from there. The driver has no scope
+of interpreting what we ask it to do.
+
+_This has a downside as well_. The GPU now accesses the main memory over PCIe.
+So if we're going to reuse the same data in the next frame, it has to go through
+the PCIe bottleneck again. But, we never use the same data again and hence gain
+a big performance boost.
+
+[wwylele](https://github.com/wwylele) tweaked the configuration UI and renamed
+some parts of it. However, as users became accustomed to the previous UI (while
+it was still in Canary) many of them were confused on where the earlier options went.
+
+#### **What's changed in the UI?**
+
+With the initial release of GLvtx, we've had support for running shaders on the
+host GPU . So, we had a dropdown menu called **Shader Emulation** to select either
+CPU or GPU. The option has now been renamed as **Enable Hardware Shader**. When
+enabled, it makes use of host GPU and when disabled, it uses CPU.
+
+Now as many of the users were using older hardware, we introduced an option called
+**Accurate Hardware Shader**. It was later renamed to **Accurate Multiplication**,
+because technically that's what it was doing. ***When enabled it would render games
+with more accuracy at the cost of slower performance***. Its the same with
+**Accurate Geometry Shader**. Earlier it was configurable via the ini file, but
+now its a check box to enable or disable accurate geometry shader.
+
+&nbsp;
+{{< sidebyside "image" "/images/entry/citra-progress-report-2018-q1/"
+    "renderer_settings.png=Old Graphics Tab"
+    "renderer_settings_new.png=New Graphics Tab" >}}
+
+During its time in Canary, a lot of testing was done by our contributors.
+They tested many number of games, under varied hardware conditions and submitted
+us with valuable telemetry data, crash logs, etc., which helped us in fixing
+issues with the hardware shader support.
+
+**Note**:
+Unfortunately, due to driver issues with AMD GPUs and inadequate OpenGL support on
+macOS,the hardware shader might not perform better than the CPU renderer under
+these conditions. However, worry not, as research is being done into possible fixes
+for these and we might have them sooner than you think.
```

Comment:
This is awkward, maybe "research is being done into possible fixes that might come around sooner than you might think."